qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferData.h

```diff
@@ -0,0 +1,285 @@
+             const IETGroupingData &groupingData, const IETBackgroundData &backgroundData,
+             const IETAnalysisData &analysisData, const IETRebinData &rebinData, const IETOutputData &outputData)
+      : m_inputData(inputData), m_conversionData(conversionData), m_groupingData(groupingData),
+        m_backgroundData(backgroundData), m_analysisData(analysisData), m_rebinData(rebinData),
+        m_outputData(outputData) {}
+
+  IETInputData getInputData() const { return m_inputData; }
+  IETConversionData getConversionData() const { return m_conversionData; }
+  IETGroupingData getGroupingData() const { return m_groupingData; }
+  IETBackgroundData getBackgroundData() const { return m_backgroundData; }
+  IETAnalysisData getAnalysisData() const { return m_analysisData; }
+  IETRebinData getRebinData() const { return m_rebinData; }
+  IETOutputData getOutputData() const { return m_outputData; }
+
+private:
+  IETInputData m_inputData;
+  IETConversionData m_conversionData;
+  IETGroupingData m_groupingData;
+  IETBackgroundData m_backgroundData;
+  IETAnalysisData m_analysisData;
+  IETRebinData m_rebinData;
+  IETOutputData m_outputData;
+};
+
+class IETPlotData {
+public:
+  IETPlotData(const IETInputData &inputData, const IETConversionData &conversionData,
+              const IETBackgroundData &backgroundData)
+      : m_inputData(inputData), m_conversionData(conversionData), m_backgroundData(backgroundData) {}
+
+  IETInputData getInputData() const { return m_inputData; }
+  IETConversionData getConversionData() const { return m_conversionData; }
+  IETBackgroundData getBackgroundData() const { return m_backgroundData; }
+
+private:
+  IETInputData m_inputData;
+  IETConversionData m_conversionData;
+  IETBackgroundData m_backgroundData;
+};
+
+class IETSaveData {
+public:
+  IETSaveData(const bool &nexus = false, const bool &spe = false, const bool &nxspe = false, const bool &ascii = false,
+              const bool &aclimax = false, const bool &daveGrp = false)
+      : m_nexus(nexus), m_spe(spe), m_nxspe(nxspe), m_ascii(ascii), m_aclimax(aclimax), m_daveGrp(daveGrp) {}
+
+  bool getNexus() const { return m_nexus; }
+  bool getSPE() const { return m_spe; }
+  bool getNXSPE() const { return m_nxspe; }
+  bool getASCII() const { return m_ascii; }
+  bool getAclimax() const { return m_aclimax; }
+  bool getDaveGrp() const { return m_daveGrp; }
+
+private:
+  bool m_nexus;
+  bool m_spe;
+  bool m_nxspe;
+  bool m_ascii;
+  bool m_aclimax;
+  bool m_daveGrp;
+};
+
+class IETGroupOption {
+public:
+  static inline const std::string UNGROUPED = "Ungrouped";
+  static inline const std::string GROUP = "Grouped";
+  static inline const std::string SAMPLECHANGERGROUPED = "Sample changer grouped";
+};
+
+class IETRebinType {
+public:
+  static inline const std::string SINGLE = "Single";
+  static inline const std::string MULTIPLE = "Multiple";
+};
+
+class IETGroupingConstants {
```

Comment:
can we have a contents at the top of the file as a doc string (e.g. class A line 231), so its easy for developers to find a class.

@RichardWaiteSTFC said it would also be good to have a brief description for why we have each one